components/admin_console/team_channel_settings/channel/details/index.ts

```diff
@@ -1,82 +1,94 @@
 
-import {bindActionCreators} from 'redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
 import {getChannel, getChannelModerations} from 'mattermost-redux/selectors/entities/channels';
 import {getAllGroups, getGroupsAssociatedToChannel} from 'mattermost-redux/selectors/entities/groups';
 import {getScheme} from 'mattermost-redux/selectors/entities/schemes';
 import {getScheme as loadScheme} from 'mattermost-redux/actions/schemes';
 import {
+    addChannelMember,
+    deleteChannel,
     getChannel as fetchChannel,
+    getChannelModerations as fetchChannelModerations,
     membersMinusGroupMembers,
     patchChannel,
-    updateChannelPrivacy,
-    getChannelModerations as fetchChannelModerations,
     patchChannelModerations,
-    addChannelMember,
     removeChannelMember,
-    updateChannelMemberSchemeRoles,
-    deleteChannel,
     unarchiveChannel,
+    updateChannelMemberSchemeRoles,
+    updateChannelPrivacy,
 } from 'mattermost-redux/actions/channels';
 import {getTeam as fetchTeam} from 'mattermost-redux/actions/teams';
 
 import {
     getGroupsAssociatedToChannel as fetchAssociatedGroups,
     linkGroupSyncable,
-    unlinkGroupSyncable,
     patchGroupSyncable,
+    unlinkGroupSyncable,
 } from 'mattermost-redux/actions/groups';
 
 import {connect} from 'react-redux';
-
 import {getTeam} from 'mattermost-redux/selectors/entities/teams';
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import {ActionFunc} from 'mattermost-redux/types/actions';
 
 import {setNavigationBlocked} from 'actions/admin_actions';
 
-import ChannelDetails from './channel_details';
+import ChannelDetails, {ChannelDetailsActions, ChannelDetailsProps} from './channel_details';
 
-function mapStateToProps(state, props) {
+function mapStateToProps(state: GlobalState, props: ChannelDetailsProps) {
     const config = getConfig(state);
     const guestAccountsEnabled = config.EnableGuestAccounts === 'true';
-    const channelID = props.match.params.channel_id;
+    const channelID = props.channelID;
     const channel = getChannel(state, channelID) || {};
     const team = getTeam(state, channel.team_id) || {};
     const groups = getGroupsAssociatedToChannel(state, channelID);
-    const allGroups = getAllGroups(state, channel.team_id);
     const totalGroups = groups.length;
+    const allGroups = getAllGroups(state);
     const channelPermissions = getChannelModerations(state, channelID);
     const teamScheme = getScheme(state, team.scheme_id);
     return {
+        channelID,
         channel,
         team,
-        allGroups,
-        totalGroups,
         groups,
-        channelID,
+        totalGroups,
+        allGroups,
         channelPermissions,
         teamScheme,
         guestAccountsEnabled,
     };
 }
 
-function mapDispatchToProps(dispatch) {
+function setNavigationBlockedFunc(blocked: boolean): ActionFunc {
```

Comment:
Not a huge fan of this, what's the error you're getting without this?

Reply:
`TS2322: Type '(blocked: any) => { type: "SET_NAVIGATION_BLOCKED"; blocked: any; }' is not assignable to type 'ActionCreator<ActionFunc>'.   Type '{ type: "SET_NAVIGATION_BLOCKED"; blocked: any; }' is not assignable to type 'ActionFunc'.     Type '{ type: "SET_NAVIGATION_BLOCKED"; blocked: any; }' provides no match for the signature '(dispatch: DispatchFunc, getState: GetStateFunc): { data: any; } | { error: any; } | Promise<{ data: any; } | { error: any; } | ActionResult[]>'.`

Reply:
Did you try doing an `as any` on the end of the function in the `ActionCreatorsMapObject`? Or just have it return `any`?

Reply:
ok, fixed in f76b723